lib/fizzy/parser_expr.cpp

```diff
@@ -209,6 +209,9 @@ parser_result<Code> parse_expr(const uint8_t* pos, const uint8_t* end)
 
         case Instr::block:
         {
+            if (pos == end)
+                throw parser_error{"Unexpected EOF"};
+
             const uint8_t type{*pos};
```

Comment:
Perhaps it would be due having a helper for the block/valtype because this code is repeated three times: block, loop, if

Reply:
Done in #174.